docs/adrs/0549-composite-run-steps.md

```diff
@@ -0,0 +1,273 @@
+TODO: Change file name to represent the correct PR number (PR is not created yet for this ADR)
+
+# ADR 0549: Composite Run Steps
+
+**Date**: 2020-06-17
+
+**Status**: Proposed
+
+**Relevant PR**: https://github.com/actions/runner/pull/549
+
+## Context
+Customers want to be able to compose actions from actions (ex: https://github.com/actions/runner/issues/438)
+
+An important step towards meeting this goal is to build in functionality for actions where users can simply execute any number of steps. 
+
+## Decision
+In this ADR, we support running multiple steps in an Action. In doing so, we build in support for mapping and flowing the inputs, outputs, and env variables (ex: All nested steps should have access to its parents' input variables and nested steps can overwrite the input variables).
+
+### Steps
+Example `user/composite/action.yml`
+```
+...
+using: 'composite' 
+steps:
+  - run: pip install -r requirements.txt
+  - run: npm install
+```
+Example `workflow.yml`
+```
+...
+jobs:
+  build:
+    runs-on: self-hosted
+    steps:
+    - id: job1
+      uses: actions/setup-python@v1
+    - id: job2
+      uses: actions/setup-node@v2
+    - uses: actions/checkout@v2
+      needs: [job1, job2]
```

Comment:
I don't think we support a step level needs: https://github.com/actions/runner/blob/master/src/Sdk/DTPipelines/workflow-v1.0.json

Is this a new addition as a part of this adr?